sql/core/src/main/scala/org/apache/spark/sql/internal/SessionState.scala

```diff
@@ -65,12 +80,29 @@ private[sql] class SessionState(sparkSession: SparkSession) {
     hadoopConf
   }
 
-  lazy val experimentalMethods = new ExperimentalMethods
+  lazy val experimentalMethods: ExperimentalMethods = {
+    existingSessionState
+      .map(_.experimentalMethods.clone)
+      .getOrElse(new ExperimentalMethods)
+  }
 
   /**
    * Internal catalog for managing functions registered by the user.
    */
-  lazy val functionRegistry: FunctionRegistry = FunctionRegistry.builtin.copy()
+  lazy val functionRegistry: FunctionRegistry = {
```

Comment:
It's better to just add a `copy` method to `FunctionRegistry` to simplify these codes.